crates/db_views/src/registration_applications/registration_application_view.rs

```diff
@@ -13,98 +14,90 @@ use lemmy_db_schema::{
-      ))
-  };
-
-  let read = move |mut conn: DbConn<'a>, search: ReadBy| async move {
-    let mut query = all_joins(registration_application::table.into_boxed());
-
-    query = match search {
-      ReadBy::Id(id) => query.filter(registration_application::id.eq(id)),
-      ReadBy::Person(person_id) => query.filter(person::id.eq(person_id)),
-    };
-
-    query.first(&mut conn).await
-  };
-
-  let list = move |mut conn: DbConn<'a>, o: RegistrationApplicationQuery| async move {
-    let mut query = all_joins(registration_application::table.into_boxed());
-
-    // If viewing all applications, order by newest, but if viewing unresolved only, show the oldest
-    // first (FIFO)
-    if o.unread_only {
-      query = query
-        .filter(registration_application::admin_id.is_null())
-        .order_by(registration_application::published.asc());
-    } else {
-      query = query.order_by(registration_application::published.desc());
-    }
-
-    if o.verified_email_only {
-      query = query.filter(local_user::email_verified.eq(true))
-    }
-
-    let (limit, offset) = limit_and_offset(o.page, o.limit)?;
-
-    query = query.limit(limit).offset(offset);
-
-    query.load::<RegistrationApplicationView>(&mut conn).await
-  };
+#[diesel::dsl::auto_type]
+/// A missing admin id, means the application is unread
+fn is_unread() -> _ {
+  registration_application::admin_id.is_null()
+}
 
-  Queries::new(read, list)
+type SelectionType = (
+  <registration_application::table as diesel::Table>::AllColumns,
+  <local_user::table as diesel::Table>::AllColumns,
+  <person::table as diesel::Table>::AllColumns,
+  Nullable<(
+    AliasedField<aliases::Person1, person::id>,
+    AliasedField<aliases::Person1, person::name>,
+    AliasedField<aliases::Person1, person::display_name>,
+    AliasedField<aliases::Person1, person::avatar>,
+    AliasedField<aliases::Person1, person::banned>,
+    AliasedField<aliases::Person1, person::published>,
+    AliasedField<aliases::Person1, person::updated>,
+    AliasedField<aliases::Person1, person::actor_id>,
+    AliasedField<aliases::Person1, person::bio>,
+    AliasedField<aliases::Person1, person::local>,
+    AliasedField<aliases::Person1, person::private_key>,
+    AliasedField<aliases::Person1, person::public_key>,
+    AliasedField<aliases::Person1, person::last_refreshed_at>,
+    AliasedField<aliases::Person1, person::banner>,
+    AliasedField<aliases::Person1, person::deleted>,
+    AliasedField<aliases::Person1, person::inbox_url>,
+    AliasedField<aliases::Person1, person::matrix_user_id>,
+    AliasedField<aliases::Person1, person::bot_account>,
+    AliasedField<aliases::Person1, person::ban_expires>,
+    AliasedField<aliases::Person1, person::instance_id>,
+  )>,
+);
+
+fn selection() -> SelectionType {
+  (
+    registration_application::all_columns,
+    local_user::all_columns,
+    person::all_columns,
+    aliases::person1.fields(person::all_columns).nullable(),
+  )
```

Comment:
When aliased fields are involved, we can't use `const`, and also can't simplify the aliased select types. Oh well.

Reply:
The proper way around that is to not have a free standing `selection()` function but to use `#[derive(Selectable)]` + '.select(Type::as_select())` instead.

Reply:
I tried that, but it doesn't work here because its not a single table we're selecting from, but the results from several joined tables.